src/components/hooks.js

```diff
@@ -190,36 +190,32 @@ export function useGasPrice(enabled = true) {
       const run = async () => {
         const provider = await getProvider()
         const blockDetails = await provider.getBlock('latest')
-        const baseFeeWei = utils.formatUnits(blockDetails.baseFeePerGas, 'wei')
-        const price = {
-          slow: baseFeeWei + 2 * Math.pow(10, 9),
-          fast: baseFeeWei * 2 + 2 * Math.pow(10, 9)
+        if (blockDetails.baseFeePerGas) {
+          const baseFeeWei = utils.formatUnits(
+            blockDetails.baseFeePerGas,
+            'wei'
+          )
+          const price = {
+            slow: baseFeeWei + 2 * Math.pow(10, 9),
+            fast: baseFeeWei * 1.5 + 2 * Math.pow(10, 9)
+          }
+          setPrice(price)
+          setLoading(false)
+        } else {
+          const gasApi = 'https://www.gasnow.org/api/v3/gas/price'
+          const result = await fetch(gasApi)
+          if (!result.ok) throw 'Failed to get gas estimate'
```

Comment:
Maybe `throw` with reason included? Something like; 
`` result.text().then(reason => throw Error(`Failed to get gas estimate. Reason: ${reason}`)) ``

Reply:
Yes good point